src/hook/useChat.ts

```diff
@@ -0,0 +1,78 @@
+import { useState } from 'react';
+import { useAlert } from './useAlert';
+
+// 상태 객체의 타입 정의
+interface Chat {
+  id: string;
+  time: string;
+  value: string;
+}
+
+// 반환되는 훅의 타입 정의
+interface UseChatReturn {
+  inputValue: string;
+  setInputValue: React.Dispatch<React.SetStateAction<string>>;
+  sendChat: () => void;
+  inputChatList: Chat[];
+  handleBack: () => void;
+  alertBack: () => void;
+  alertVideo: () => void;
+}
+
+export function useChat(): UseChatReturn {
+  const [inputValue, setInputValue] = useState<string>('');
+  const [createObject, setCreateObject] = useState<Chat>({
+    id: 'eonny',
+    time: new Date().toLocaleTimeString([], {
+      hour: '2-digit',
+      minute: '2-digit',
+    }), // 초를 제외한 시간 설정
+    value: '',
+  });
+  const [inputChatList, setInputChatList] = useState<Chat[]>([]);
+
+  // alert 훅 호출 Test
+  const handleBack = () => {
+    alert('이것은 다음 과제🦖');
+  };
+
+  const { alertBack, alertVideo } = useAlert();
+
+  const sendChat = () => {
+    if (inputValue.trim()) {
+      // 새로운 채팅 객체 생성
+      const newChat: Chat = {
+        ...createObject,
+        time: new Date().toLocaleTimeString([], {
+          hour: '2-digit',
+          minute: '2-digit',
+        }), // 초를 제외한 시간으로 갱신
+        value: inputValue, // 입력된 값을 객체의 value로 설정
+      };
+
+      // 새로운 객체를 리스트에 추가
+      setInputChatList([...inputChatList, newChat]);
```

Comment:
`spread` 연산자 사용 아주 좋습니다! 다만 `setState`를 사용할 때 변경되는 state를 직접 참조하여 사용하는 것보다는 콜백 함수를 작성하는 방식이 조금 더 안정적인 방식입니다 🙂

```suggestion
      setInputChatList((prev) => [...prev, newChat]);
```

---

프로젝트를 진행하다 보면 `state 변경 -> 렌더링 -> state 변경 -> 렌더링 -> ...` 과정을 반복하게 됩니다.

`setState(state + 1)`은 현재의 state 값을 기반으로 상태를 업데이트 합니다. 하지만 이 방식은 state 값이 변경되는 과정에서 다른 setState 호출이 있을 경우, 이전 렌더링 시점의 상태를 기반으로 하기 때문에 최신 상태값을 반영하지 못할 수 있습니다.

반면 `setState(prev => prev + 1)`은 콜백 함수를 사용하여 이전 상태(prev)를 기준으로 상태를 업데이트 합니다. 이 방법은 상태 업데이트가 비동기적으로 처리되는 동안에도 항상 최신 상태를 반영하므로, 상태가 여러 번 업데이트 될 때에도 정확한 결과를 보장한다고 합니다!